src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,432 @@
+     * @return the instantiated object that will be used by the command line parser
+     * as an argument source
+     * @throws IllegalAccessException
+     * @throws InstantiationException
+     */
+    //TODO
+    @Override
+    public Object getInstance(Class<?> pluggableClass) throws IllegalAccessException, InstantiationException {
+        Annotation annot = null;
+        final String simpleName = pluggableClass.getSimpleName();
+
+        if (allDiscoveredAnnotations.containsKey(simpleName)) {
+            // we found a plugin class with a name that collides with an existing class;
+            // plugin names must be unique even across packages
+            throw new IllegalArgumentException(
+                    String.format("A plugin class name collision was detected (%s/%s). " +
+                                    "Simple names of plugin classes must be unique across packages.",
+                            pluggableClass.getName(),
+                            allDiscoveredAnnotations.get(simpleName).getClass().getName())
+            );
+        } else if (toolDefaultAnnotations.containsKey(simpleName)) {
+            // an instance of this class was provided by the tool as one of it's default filters;
+            // use the default instance as the target for command line argument values
+            // rather than creating a new one, in case it has state provided by the tool
+            annot = toolDefaultAnnotations.get(simpleName);
+        } else {
+            annot = (Annotation) pluggableClass.newInstance();
+        }
+
+        // Add all filters to the allDiscoveredReadFilters list, even if the instance came from the
+        // tool defaults list (we want the actual instances to be shared to preserve state)
+        allDiscoveredAnnotations.put(simpleName, annot);
+
+        Class<?>[] interfaces = annot.getClass().getInterfaces();
+        for (Class<?> inter : interfaces) {
+            // Following with how groups are currently defined and discovered, namely they are interfaces that
+            // extend Annotation. This might not be entirely desirable as it results in ReducibleAnnotation being
+            // a definable group even though this is not an entirely helpful list.
+            if ((inter != pluginBaseClass) && (pluginBaseClass.isAssignableFrom(inter))) {
+                List<Annotation> list = discoveredGroups.containsKey(inter.getSimpleName()) ? discoveredGroups.get(inter.getSimpleName()) : new ArrayList<>();
+                list.add(annot);
+                discoveredGroups.put(inter.getSimpleName(), list);
+
+                // If its a valid group, check whether the tool requested that group and add it to default annotations
+                if (toolDefaultGroups.contains(inter.getSimpleName()) && !toolDefaultAnnotations.containsKey(simpleName)) {
+                    toolDefaultAnnotations.put(simpleName, annot);
+                }
+            }
+        }
```

Comment:
^I have cleaned up the comments.
@cmnbroad Actually no, because the interfaces are dynamically discovered from the implementing annotations we never actually get handed a `ReducibleAnnotaiton` object in `getInstance()`. If we were it would be possible to filter it at the getClassFilter but it appears where `getInstance()` is called it just adds the return value directly into its running list of instantiated objects. We wouldn't want that to be the case because the annotation groups are interfaces and thus can't be directly instantiated.